packages/datadog-plugin-mocha/src/index.js

```diff
@@ -38,6 +38,7 @@ function createWrapRunTest (tracer, testEnvironmentMetadata, sourceRoot) {
   return function wrapRunTest (runTest) {
     return async function runTestWithTrace () {
       let specFunction = this.test.fn
+      const context = this.test.ctx
```

Comment:
Is it expected that this is assigned before `this.test.fn`? Can `this.test.fn` be called only once? Otherwise wouldn't it be possible that it ends up with a previous context?

Reply:
the context is kept in retries:
https://github.com/mochajs/mocha/blob/master/lib/runner.js#L816-L824 https://github.com/mochajs/mocha/blob/c667d1060f97da24abf48e41a284e037f0b410a0/lib/test.js#L81, which I think should be the only time the spec function is called more than once

Reply:
do you have any recommendation in mind? Or would we be good with this?